messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/BridgeAdd.java

```diff
@@ -166,48 +179,77 @@ static BridgeConfiguration createBridgeConfiguration(final OperationContext cont
         }
         return result;
     }
 
-    static void createBridge(String name, BridgeConfiguration bridgeConfig, ActiveMQServerControl serverControl) {
+    static void createBridge(BridgeConfiguration bridgeConfig, ActiveMQServer server) {
+        checkStarted(server);
+        clearIO(server);
+
         try {
-            String transformerClassName = bridgeConfig.getTransformerConfiguration() != null ? bridgeConfig.getTransformerConfiguration().getClassName() : null;
-            if (bridgeConfig.getDiscoveryGroupName() != null) {
-                serverControl.createBridge(name, bridgeConfig.getQueueName(), bridgeConfig.getForwardingAddress(),
-                        bridgeConfig.getFilterString(), transformerClassName, bridgeConfig.getRetryInterval(),
-                        bridgeConfig.getRetryIntervalMultiplier(), bridgeConfig.getInitialConnectAttempts(),
-                        bridgeConfig.getReconnectAttempts(), bridgeConfig.isUseDuplicateDetection(),
-                        bridgeConfig.getConfirmationWindowSize(), bridgeConfig.getClientFailureCheckPeriod(),
-                        bridgeConfig.getDiscoveryGroupName(), true, bridgeConfig.isHA(), bridgeConfig.getUser(),
-                        bridgeConfig.getPassword());
-            } else {
-                boolean first = true;
-                String connectors = "";
-                for (String connector : bridgeConfig.getStaticConnectors()) {
-                    if (!first) {
-                        connectors += ",";
-                    } else {
-                        first = false;
-                    }
-                    connectors += connector;
-                }
-                serverControl.createBridge(name, bridgeConfig.getQueueName(), bridgeConfig.getForwardingAddress(),
-                        bridgeConfig.getFilterString(), transformerClassName, bridgeConfig.getRetryInterval(),
-                        bridgeConfig.getRetryIntervalMultiplier(), bridgeConfig.getInitialConnectAttempts(),
-                        bridgeConfig.getReconnectAttempts(), bridgeConfig.isUseDuplicateDetection(),
-                        bridgeConfig.getConfirmationWindowSize(), bridgeConfig.getClientFailureCheckPeriod(),
-                        connectors, false, bridgeConfig.isHA(), bridgeConfig.getUser(),
-                        bridgeConfig.getPassword());
-            }
+            server.deployBridge(bridgeConfig);
         } catch (RuntimeException e) {
             throw e;
         } catch (Exception e) {
             // TODO should this be an OFE instead?
             throw new RuntimeException(e);
+        } finally {
+            blockOnIO(server);
+        }
+    }
+
+    static void checkStarted(ActiveMQServer server) {
+        // extracted from ActiveMQServerControlImpl#checkStarted()
+
+        if (!server.isStarted()) {
+            throw new IllegalStateException("Broker is not started. It can not be managed yet");
```

Comment:
@TomasHofman this needs to be an i18n message

Reply:
also s/can not/cannot

Reply:
Thanks @bstansberry , fixed.